### PR TITLE
DF 1690 - add Feign Client for candidate service and fetching a Candi…

### PR DESCRIPTION
**Pull request by emina-str**
…date by PersonNumber

- remove dependence of BU-JobCenterCode from ProofOfWorkEfforts (NpA) to CandidateFeignClient

CandidateService : https://github.com/alv-ch/candidate-service/pull/138

### CODE REVIEW

File: infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/feign/CandidateProfileDto.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package ch.admin.seco.alv.service.onlineform.infrastructure.feign;
+
+import java.util.UUID;
+
+public class CandidateProfileDto {
+
+    private UUID id;
```
Comment (pavax) [inline infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/feign/CandidateProfileDto.java]: change it to a string

File: infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/messagebroker/events/DomainEventSenderGateway.java [hunk 2]
```diff
@@ -4,9 +4,12 @@
 import ch.admin.seco.alv.service.onlineform.application.bu.ApplicationDocumentDto;
 import ch.admin.seco.alv.service.onlineform.application.npa.ProofOfWorkEffortsApplicationService;
 import ch.admin.seco.alv.service.onlineform.application.npa.ProofOfWorkEffortsDto;
+import ch.admin.seco.alv.service.onlineform.domain.PersonNumber;
 import ch.admin.seco.alv.service.onlineform.domain.bu.events.ApplicationDocumentDeletedEvent;
 import ch.admin.seco.alv.service.onlineform.domain.bu.events.ApplicationDocumentEvent;
 import ch.admin.seco.alv.service.onlineform.domain.npa.events.ProofOfWorkEffortsSubmittedEvent;
+import ch.admin.seco.alv.service.onlineform.infrastructure.feign.CandidateFeignClient;
```
Comment (pavax) [inline infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/messagebroker/events/DomainEventSenderGateway.java]: a infrastructure package must not reference another intrastructure package  (see onion-architecture).  They should either be in the same package or even better use a ApplicationService.

In our case I'd suggest to:
Create a method findJobCenterCode(PersonNumber personNumber) in the UserService.java.

File: infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/feign/CandidateProfileDto.java (hunk 1 above)
Comment (emina-str) [reply infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/feign/CandidateProfileDto.java]: fixed.

File: infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/messagebroker/events/DomainEventSenderGateway.java (hunk 2 above)
Comment (emina-str) [reply infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/messagebroker/events/DomainEventSenderGateway.java]: added method `Optional<String> findJobCenterCode(PersonNumber personNumber)` in UserService.

**Pull request merged**